### PR TITLE
Update Github Workflows

### PR DESCRIPTION
**Pull request by mbollmann**
1. Updates `actions/checkout` to `@v3` because older versions are raising a deprecation warning due to Node.js 12.
2. Replaces `::set-output` commands because [they are deprecated and will start failing on June 1st, 2023](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).
3. Adds a check if we're in the `acl-org/acl-anthology` repo to the `preview` and `publish` workflows, which will stop Github from attempting to run them in forks of this repo (which fails, but is annoying).

### CODE REVIEW

**Review by mjpost — APPROVED** [review]
Thanks for noticing this! I approved it so it's ready to merge when you figure out why the preview failed...

**Comment by mbollmann** [comment]
I can't see anything wrong with it, and re-running it in debug mode made it succeed, so I guess it was a random hiccup?

Github Workflows are a bit hard to test in advance, so let's keep a close eye on them after merging...

**Pull request merged**
